### ISSUE TITLE
Cannot manually scale GUI on Linux

### ISSUE THREAD

**Issue by BartMassey** [emilk/egui#1721]
Issue #53 describes the need to be able to manually scale the GUI both for accessibility and for dealing with HIDPI displays. The GUI should be scalable using `<ctl-shift-"+">` and `<ctl-shift-"-">` or `<ctl-mousewheel>`, . Neither of these options currently works on Linux.

This can be reproduced by compiling the `hello world` demo app and trying to scale it.

I'm running current Debian Linux.

**Comment by emilk** [comment]
This should already work well on the web (please www.egui.rs).

In case you are talking about native:
* are you used to be able to zoom any app like that?
* have you tested just listening for those key presses and calling `egui_ctx.set_pixels_per_point` in your app?

**Comment by BartMassey** [comment]
Yes, native. Sorry, should have been more clear.

> are you used to be able to zoom any app like that?

Very much so.

> have you tested just listening for those key presses and calling egui_ctx.set_pixels_per_point in your app?

Nope! Could, but was hoping that the framework would handle it.

**Comment by thomas992** [comment]
What apps are you able to zoom in? My understanding is that zooming is a browser only event currently.
```    let mut ctx = egui::Context::default();
    if ui
        .input_mut()
        .consume_key(egui::Modifiers::ALT_SHIFT, Key::F)
    {
        ctx.set_pixels_per_point(120.0)
    }```

with the pub use context at the start does not work in my easymark editor branch

**Comment by BartMassey** [comment]
Gnome-terminal, for example, zooms the whole window on `<ctrl>-<minus>` and `<ctrl>-<plus>`.  I mostly edit in a terminal, so idk what editors do. VSCode seems to support `<ctrl>-<minus>` and `<ctrl>-<plus>`, although tbf it uses some browser framework underneath I'm sure. Zulip works, but again… Other apps seem to support zooming on a per-app basis: for example LibreOffice Calc zooms the spreadsheet with `<ctrl>-<mousewheel>`.

**Comment by thomas992** [comment]
Accessing the minus and plus buttons are not possible on EGUI. Keys such as ; are not international and requires some form of internationalization to make accessible in EGUI.

**Comment by kud1ing** [comment]
> This should already work well on the web (please [www.egui.rs](http://www.egui.rs/)).

On Safari macOS this seems to zoom into the canvas, not scaling the rendering within the canvas, though.

<img width="669" alt="Bildschirmfoto 2022-06-18 um 10 33 51" src="https://user-images.githubusercontent.com/391975/174429812-08b8d4fc-44b0-436b-9067-b9aec863377f.png">
